content/blog/managing-google-workspace-with-terraform.md

```diff
@@ -0,0 +1,295 @@
+- [Wrapping Up](#wrapping-up)
+
+## Growing Pains
+
+As Masterpoint has been around for almost 10 years now, we've experienced growing pains transitioning from a one-person consultancy to a small team of core full-time engineers, with the help of additional contractors when needed.
+
+During that time, we've focused on client projects more than our own internal systems. When I joined in December, we started to feel the friction of onboarding a new Google Workspace user more acutely. Matt (CEO/CTO) had to remember what permissions a new Google Workspace user gets by default, how to provision SSO permissions for Masterpoint's AWS Accounts, and how to get a user set up for client-specific SSOs. Since he hadn't gone through that process in 6 months, he forgot some important details, which resulted in delayed access to the Masterpoint AWS accounts and took time away from his ability to focus on other work due to tedious administrative tasks.
+
+## Why This Matters for Scaling Teams
+
+In our small company and possibly in your own organization, it's not a big deal for the founder or early engineer to create a new employee's Google Workspace account or give them SSO account access via the Admin UI. But as a company scales:
+
+- Founders or engineers are (hopefully) focused on product, customers, and strategy.
+- Nobody remembers why or when a permission was changed.
+- Group membership becomes inconsistent.
+- Onboarding/offboarding gets slower and error-prone.
+- The stakes for getting security permissions correct increase.
+
+As a company grows past 10, dozens, or 100+ people, teams (and dare we say departments) become more distributed and harder to manage across many different SaaS tools. It's important to have systems in place that provide clarity and enable individuals to get the right work done in a timely manner. We believe efficient systems are part of a company evolving into higher levels of organizational maturity.
+
+## Managing Google Workspace with Terraform
+
+We looked at a couple of open source solutions and decided on using the Terraform [Google Workspace](https://github.com/hashicorp/terraform-provider-googleworkspace) provider. We often use SaaS-specific Terraform providers to provision user accounts for DataDog and other SaaS services, so this was an easy decision.
+
+We heard great things about [GAM](https://github.com/GAM-team/GAM) (an imperative command line tool) from a few colleagues, but we didn't need its full range of capabilities and prefer to use declarative systems so we know the user account and group settings we see in config files are indeed the values in production.
+
+In terms of using the provider, while there are a few Terraform modules out there for managing Google Workspaces, we decided to create our own Terraform module. The motivation to build a new module from scratch stemmed from our desire to:
+
+1. Democratize user onboarding powers so Matt (CEO/CTO) is not a bottleneck.
```

Comment:
These are all good things, but wouldn't they be available with any module? Was there some reason you needed to write a module from scratch to achieve these?

Also, you already mentioned Matt is the CEO/CTO, so I'd remove that parenthetical.

Reply:
That's good feedback. From a pure engineering focus, 3 and 4 were the motivations for building our own module. Other modules ignored and overrode the SSO settings and we walk through the intuitive set up in the design decision. I think I'll remove 1/2 and add another more technical motivation.